src/handlers/strategies/directoryListing.ts

```diff
@@ -111,41 +167,61 @@ export async function listDirectory(
   env: Env
 ): Promise<Response> {
   const delimitedPrefixes = new Set<string>();
-  const objects: R2Object[] = [];
+  const objects: _Object[] = []; // s3 sdk types are weird
+
+  // Create an S3 client instance to interact with the bucket.
+  // There is a limit in the size of the response that
+  //  a binding can return. We kept hitting it due to the
+  //  size of our paths, causing us to send a lot of requests
+  //  to R2 which in turn added a lot of latency. The S3 api
+  //  doesn't have that response body size constraint so we're
+  //  using it for now.
+  const client = new S3Client({
```

Comment:
Should this instantiation not be global? How long it takes for the S3 client to be instantiated and ready to be used?

Reply:
Instantiating it takes <1ms, it's generally not recommended to put things in the global scope in a worker since iirc that makes it possible for it to be shared by multiple requests and that can get a little messy